kernel/libs/aster-bigtcp/src/iface/poll.rs

```diff
@@ -337,9 +421,14 @@ impl<E: Ext> PollContext<'_, E> {
             return did_something_tcp;
         };
 
-        let (did_something_udp, _tx_token) = self.dispatch_udp(tx_token, dispatch_phy);
+        let (did_something_udp, tx_token) = self.dispatch_udp(tx_token, dispatch_phy);
+        let Some(tx_token) = tx_token else {
+            return did_something_udp;
+        };
+
+        let (did_something_raw, _tx_token) = self.dispatch_raw(tx_token, dispatch_phy);
 
-        did_something_tcp || did_something_udp
+        did_something_tcp || did_something_udp || did_something_raw
     }
 
     fn dispatch_tcp<T, Q>(&mut self, tx_token: T, dispatch_phy: &mut Q) -> (bool, Option<T>)
```

Comment:
For TCP sockets, `dispatch` also processes generated responses:
https://github.com/asterinas/asterinas/blob/d958ea3153935315d630b7a80324fc4cf8ff8a7b/kernel/libs/aster-bigtcp/src/socket/bound.rs#L991-L999

We need to ensure that each packet can be captured by raw sockets once and only once, so we need to make some adjustments to this logic.